lib/test.js

```diff
@@ -58,8 +58,16 @@ function configure(sinon, config) {
             } catch (e) {
                 exception = e;
             }
-
-            if (typeof oldDone !== "function") {
```

Comment:
Hi!

I tried s.th. similar and came to the following code:

if (typeof oldDone != "function") {
                    if (result && typeof result.then === "function") {
                        return result.then(function (result) {
                            sandbox.verifyAndRestore();
                            return result;
                        }, function (exception) {
                            sandbox.restore();
                            throw exception;
                        });
                    }
                    if (typeof exception !== "undefined") {
                        sandbox.restore();
                        throw exception;
                    } else {
                        sandbox.verifyAndRestore();
                    }
                }

Why? I have the feeling that this magic must only happen if (typeof oldDone !== "function"), because else oldDone has been wrapped and does the verify/restore. I also test just for "then", which accepts both a success and failure handler. This is in line with how QUnit recognizes a Promise.

What do you think?

Best regards,
     Thomas